### PR TITLE
Use the new yaml-language-server v0.11.0 ESM and UMD Modules and address tech debt

### PR DESCRIPTION
**Pull request by apisim**
This PR will implement changes to make monaco-yaml start using the ESM and UMD modules [now available](https://github.com/redhat-developer/yaml-language-server/pull/305) in yaml-language-server v0.11.0. In addition, it will address some tech debt: eliminate the patching of upstream dependencies and fix the editor formatting of YAML so it works again.

Notice that:
* The "umd" example seems to be broken. See also [this issue](https://github.com/pengx17/monaco-yaml/issues/32). This PR won't try to fix it.
* There are no unit tests but `jest` is a dependency. This PR will leave the `jest` dependencies in `package.json` even though it isn't used.
* There are [reports](https://github.com/pengx17/monaco-yaml/issues/31) that the monaco-yaml bundle size has increased significantly, presumably since the changes to remove the embedded yaml-languange-server and yaml-ast-parser-custom-tags dependencies. This PR won't try to address the issue.

### CODE REVIEW

**Comment by apisim** [comment]
Bumping up the versions of  "monaco-editor", "monaco-editor-core", and "monaco-languages" to match #35 didn't work in my tests (e.g formatting). Only the latest versions, which were  published yesterday, worked for me.

In case you are using "monaco-editor-webpack-plugin"...I'll suggest upgrading to v2.0.0+ to fix "gotoLine...not found" error.

Will someone be able to review this PR soon? I got the impression that these [changes will be welcomed](https://github.com/pengx17/monaco-yaml/pull/30#issuecomment-678454020).

Cheers!

**Comment by apisim** [comment]
Another monaco-editor version-bump. 

I hope everyone is OK in these pandemic times and can spare some time to review this PR.
Cheers!

**Review by yazaabed — APPROVED** [review]

**Comment by yazaabed** [comment]
@apisim can I merge it?

**Comment by apisim** [comment]
> @apisim Thanks man for implementing this! I tested it! Can we merge it?

Yes, I am OK with somebody with proper access merging it :100:

**Comment by yazaabed** [comment]
@fleon is it possible to publish this pr? I have no access to publish it to NPM.

**Comment by yazaabed** [comment]
@pengx17 can you please give me access to publish a new version of Monaco-yaml? My username is `yazanaabed`.

**Comment by pengx17** [comment]
@yazaabed Done!

**Comment by yazaabed** [comment]
@pengx17 thank you 👍

**Comment by yazaabed** [comment]
@pengx17 any idea why this is happening when I do publish?
```
This operation requires a one-time password.
Enter OTP:
```

**Comment by pengx17** [comment]
@yazaabed I think this npm repo requires you to setup an otp authenticator to publish a new version

**Comment by fleon** [comment]
I will create a new release today. Let me know if you would like to do it instead @yazaabed. To publish you just need to do `yarn publish` and it will automatically build everything and publish. The version we want is `2.5.1`.

@apisim Thanks a lot for your contribution.

File: README.md [hunk 1]
```diff
@@ -23,20 +23,12 @@ See `examples` directory for esm and umd examples.
...
 * `yarn`
-* open `$/monaco-yaml/demo/index.html` in your favorite browser.
 
 A running example:
 ![demo-image](test-demo.png)
 
 ## Credits
 - https://github.com/redhat-developer/yaml-language-server
 
-### Maintain
```
Comment (fleon) [inline README.md:34]: Really glad to see this section go away. 😄

**Comment by yazaabed** [comment]
@fleon @pengx17 I published the new version to npm. And tested with my employer project. It is working smoothly. Also updating to Monaco-editor latest version was really nice!! Thank you all for supporting this. Also formatting is working perfectly as expected with indentation and tabs.

**Comment by yazaabed** [comment]
Quick example:
Oct-05-2020 16-47-30

**Comment by fleon** [comment]
Looks great @yazaabed. I will try to upgrade it in GitLab next week and see if I run into any problems.

**Pull request merged**